x/proof/types/message_submit_proof_test.go

```diff
@@ -72,7 +77,11 @@ func TestMsgSubmitProof_ValidateBasic(t *testing.T) {
 				},
 				Proof: testClosestMerkleProof,
 			},
-			expectedErr: ErrProofInvalidService.Wrap("proof service ID %q cannot be empty"),
+			sessionHeaderToExpectedErrorFn: func(sh sessiontypes.SessionHeader) error {
+				serviceError := sharedtypes.ErrSharedInvalidService.Wrapf("invalid service ID: %s", sh.Service.Id)
+				sessionError := sessiontypes.ErrSessionInvalidService.Wrapf("invalid service: %v; %s", sh.Service, serviceError)
+				return ErrProofInvalidSessionHeader.Wrapf("invalid session header: %s", sessionError)
```

Comment:
:grimacing: this one is a bit trickier because `SessionHeader#ValidateBasic()` wraps `ErrSessionInvalidService` with `%v` and the service itself, followed by the service validation error. In this case, both will be empty string, which looks enigmatic when the duplicate copy is removed:

```
; : invalid service: invalid service in session: invalid session header
```

I would propose removing the `%v` usage with the service in `SessionHeader#ValidateBasic()`, removing the duplicate message copy, replace `%s` with `%q` for the service ID, and maybe apend the word "ID" to the end of `ErrSharedInvalidService`. This would result in the following error string:

```
"": invalid service ID: invalid service in session: invalid session header
```

```suggestion
				serviceError := sharedtypes.ErrSharedInvalidService.Wrapf("%q", sh.Service.Id)
				sessionError := sessiontypes.ErrSessionInvalidService.Wrapf("%s", serviceError)
				return ErrProofInvalidSessionHeader.Wrapf("%s", sessionError)
```